pyquil/gates.py

```diff
@@ -475,7 +484,7 @@ def RESET(qubit_index=None):
 """
 
 
-def MEASURE(qubit, classical_reg):
+def MEASURE(qubit: QubitDesignator, classical_reg: Optional[MemoryReferenceDesignator]) -> Measurement:
```

Comment:
The `classical_reg` arg here can also be an `int`, but that is deprecated.  The current type hint causes mypy to warn:

```
pyquil/tests/test_qvm.py:16: error: Argument 2 to "MEASURE" has incompatible type "int"; expected "Union[MemoryReference, Tuple[str, int], List[Any], str, None]"
```

This warning only pops up in the three places in `test_qvm.py`. We could update the type hint here to include the deprecated `int` type and get rid of the warnings, but it seems saner to leave it as advertising only non-deprecated types.